File: HelpfulSense/app/src/main/java/com/andruid/magic/helpfulsense/model/ContactHolder.java
package com.andruid.magic.helpfulsense.model;

import android.view.View;

import com.andruid.magic.helpfulsense.R;
import com.andruid.magic.helpfulsense.viewholder.ContactViewHolder;
import com.wafflecopter.multicontactpicker.ContactResult;

import java.util.List;
import java.util.Objects;

import eu.davidea.flexibleadapter.FlexibleAdapter;
import eu.davidea.flexibleadapter.items.AbstractFlexibleItem;
import eu.davidea.flexibleadapter.items.IFlexible;

public class ContactHolder extends AbstractFlexibleItem<ContactViewHolder> {
    private final ContactResult contact;

    public ContactHolder(ContactResult contact){
        this.contact = contact;
    }

    public ContactResult getContact() {
        return contact;
    }

    @Override
    public boolean equals(Object o) {
        if(o == null || o.getClass() != getClass())
            return false;
        ContactHolder c = (ContactHolder) o;
        return contact.equals(c.contact);
    }

    @Override
    public boolean isSwipeable() {
        return true;
    }

    @Override
    public boolean isDraggable() {
        return true;
    }

    @Override
    public int getLayoutRes() {
        return R.layout.layout_contact;
    }

    @Override
    public ContactViewHolder createViewHolder(View view, FlexibleAdapter<IFlexible> adapter) {
        return new ContactViewHolder(view, adapter);
    }

    @Override
    public void bindViewHolder(FlexibleAdapter<IFlexible> adapter, ContactViewHolder holder, int position, List<Object> payloads) {
        holder.bindContact(((ContactHolder) Objects.requireNonNull(adapter.getItem(position)))
                .contact);
    }
}

File: HelpfulSense/app/src/main/java/com/andruid/magic/helpfulsense/model/ContactHolder.kt
package com.andruid.magic.helpfulsense.model

import android.view.View
import androidx.recyclerview.widget.RecyclerView
import com.andruid.magic.helpfulsense.R
import com.andruid.magic.helpfulsense.database.entity.Contact
import com.andruid.magic.helpfulsense.ui.adapter.ContactAdapter
import com.andruid.magic.helpfulsense.ui.viewholder.ContactViewHolder
import eu.davidea.flexibleadapter.FlexibleAdapter
import eu.davidea.flexibleadapter.items.AbstractFlexibleItem
import eu.davidea.flexibleadapter.items.IFlexible

/**
 * Holder class for creating swipeable and draggable views for [ContactAdapter]
 * @property contact data for the view
 */
data class ContactHolder(
        val contact: Contact
) : AbstractFlexibleItem<ContactViewHolder>() {
    override fun bindViewHolder(adapter: FlexibleAdapter<IFlexible<RecyclerView.ViewHolder>>?,
                                holder: ContactViewHolder?, position: Int, payloads: MutableList<Any>?) {
        holder?.bind((adapter?.getItem(position) as ContactHolder).contact)
    }

    override fun createViewHolder(view: View, adapter: FlexibleAdapter<IFlexible<RecyclerView.ViewHolder>>?) =
            ContactViewHolder(view, adapter)

    override fun getLayoutRes() = R.layout.layout_contact
}

/**
 * Extension function to convert [Contact] object to [ContactHolder] for recycler view
 * @receiver contact object
 * @return contactHolder object
 */
fun Contact.toContactHolder() = ContactHolder(this)